### PR TITLE
Fix empty marker issues

### LINKED ISSUE #978: Poetry fails on keyboard add/install requirements
- [ x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [ x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [x ] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Windows 10, Python 3.7.2
- **Poetry version**: 
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: 

## Issue

Python 3.7.2
PIP 19.0.3
Setuptools 40.8.0
Poetry 1.0.0a2
######################################
Using version ^0.13.3 for keyboard

Updating dependencies
Resolving dependencies...

[InvalidRequirement]
Invalid requirement, parse error at "'; <empty'"

### PR DESCRIPTION
**Pull request by Nitori-**
Updated version of #1230 (I didn't realize rebasing would make it so I couldn't re-open the PR)

This PR addresses a family of related problems we were seeing when depending on particular packages.

Based on what I saw in the codebase, there is a concept of an "empty" marker. However there were a few places in the code that didn't handle these empty markers properly.

Here's an example `pyproject.toml` where we were hitting the issue:

```toml
[tool.poetry]
name = "example_project"
version = "1.0.0"
description = ""
authors = [""]

[tool.poetry.dependencies]
python = ">=2.7, != 3.0.*, != 3.1.*, != 3.2.*, != 3.3.*, != 3.4.*"

[tool.poetry.dev-dependencies]
pytest = "^3.0.6"
pylama = "^7.6"
pylint = { version = "^2.2", python = ">= 3.5" }
pytest-cov = "^2.6"
black = { version = "18.9b0", python = ">= 3.6" }
sphinx = { version = "^2.0", python = ">= 3.5" }

[build-system]
requires = ["poetry>=0.12"]
build-backend = "poetry.masonry.api"
```

Doing `poetry install` once to generate the `poetry.lock` and then running `poetry install` again will reproduce the issue:

```
$ poetry install
Installing dependencies from lock file

[InvalidRequirement]
Invalid requirement, parse error at "'; <empty'"

install [--no-dev] [--dry-run] [-E|--extras EXTRAS] [--develop DEVELOP]
```
Also I believe this fixes #978 which seems to be hitting a similar problem.

# Pull Request Check List

This is just a reminder about the most common mistakes. Please make sure that you tick all *appropriate* boxes.  But please read our [contribution guide](https://poetry.eustace.io/docs/contributing/) at least once, it will save you unnecessary review cycles!

- [x] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

**Note**: If your Pull Request introduces a new feature or changes the current behavior, it should be based
on the `develop` branch. If it's a bug fix or only a documentation update, it should be based on the `master` branch.

If you have *any* questions to *any* of the points above, just **submit and ask**!  This checklist is here to *help* you, not to deter you from contributing!

### CODE REVIEW

File: poetry/packages/package.py [hunk 1]
```diff
@@ -340,7 +340,7 @@ def to_dependency(self):
 
         name = "{} (=={})".format(self._name, self._version)
 
-        if not self.marker.is_any():
+        if not self.marker.is_any() and not self.marker.is_empty():
```
Comment (sdispater) [inline poetry/packages/package.py]: That's not what we want here. An empty marker is a marker that will always evaluate to false. By dropping it we make the dependency always valid which is contrary to what an empty marker means.

The issue is that we should not have empty markers make it to the lock file, which is the root cause of the problem you are seeing and this is where the underlying issue should be fixed.

Comment (Nitori-) [reply poetry/packages/package.py]: Would it be sufficient to just make sure they don't get serialized to the lock file, or is there something more fundamental about dependency resolution that needs to prune them?

**Comment by sdispater** [comment]
@Nitori- While the issue is still present on the `0.12.17` release, I can't reproduce the issue with the `1.0.0b2` release. Could you test and see if that fixes the issue for you as well.

**Comment by Nitori-** [comment]
Yes the issue appears to be gone with `1.0.0b2`, thanks @sdispater !

**Pull request closed without merging**